src/Weather_forecast_on_dashboard.py

```diff
@@ -0,0 +1,133 @@
+    if ds is None:
+        return []
+
+    interpolated_data = []
+
+    time_dim = ds['step'] if 'step' in ds.dims else ds['time'] if 'time' in ds.dims else None
+    if time_dim is None:
+        st.error("Neither 'step' nor 'time' dimension is present in the dataset. Unable to proceed with forecast.")
+        return []
+
+    for time_val in time_dim:
+        actual_forecast_time = init_datetime + pd.to_timedelta(time_val.values)
+        if actual_forecast_time.date() == forecast_date.date():
+            if parameter == "u10:v10":
+                u = ds['u10'].interp(latitude=lat, longitude=lon, method="nearest", step=time_val)
+                v = ds['v10'].interp(latitude=lat, longitude=lon, method="nearest", step=time_val)
+                value = np.sqrt(u**2 + v**2)
+            elif parameter == "u100:v100":
+                u = ds['u100'].interp(latitude=lat, longitude=lon, method="nearest", step=time_val)
+                v = ds['v100'].interp(latitude=lat, longitude=lon, method="nearest", step=time_val)
+                value = np.sqrt(u**2 + v**2)
+            else:  # temp
+                value = ds['t2m'].interp(latitude=lat, longitude=lon, method="nearest", step=time_val) - 273.15
+
+            interpolated_data.append({
+                'date_time': actual_forecast_time,
+                'latitude': lat,
+                'longitude': lon,
+                'value': value.compute().item(),
+                'init_time': f"{init_time} hours before"
+            })
+
+    return interpolated_data
+
+def get_forecast(forecast_date, lat, lon, parameter, init_times, model="ifs"):
+    all_data = []
+    with ThreadPoolExecutor() as executor:
+        results = list(executor.map(lambda init_time: process_initialization(init_time, forecast_date, lat, lon, parameter, model), init_times))
+    for result in results:
+        all_data.extend(result)
+    return pd.DataFrame(all_data)
+
+def weather_forecast_page():
+    # Streamlit app
+    st.title("Weather Forecasts from Different Initialization Times")
+
+    parameter_mapping = {
+        "Wind Speed (10m)": "u10:v10",
+        "Wind Speed (100m)": "u100:v100",
+        "Temperature": "2t"
+    }
+
+    parameter_option = st.sidebar.selectbox("Select Parameter", options=list(parameter_mapping.keys()))
+    parameter_value = parameter_mapping[parameter_option]
+    lat = st.sidebar.number_input("Enter Latitude", value=27.035, format="%.6f")
+    lon = st.sidebar.number_input("Enter Longitude", value=70.515, format="%.6f")
+
+    # Use datetime.combine to set the forecast_date to midnight
+    forecast_date = st.sidebar.date_input("Select Forecast Date", datetime.today().date())
+    forecast_date = datetime.combine(forecast_date, time(hour=0))
+
+    # Multiselect for initialization times in hours
+    init_time_options = st.sidebar.multiselect(
+        "Select Initialization Times (in hours before forecast date)",
```

Comment:
perhaps rename this to Hours before Forecast date

or re work the code, and have options to actually quick on datetimes, of when the forecast is made. This will make it clearer to run